### PR TITLE
feat: migrate to github actions

### PR DESCRIPTION
**Pull request by yohanelly95**
# Migration to GitHub Actions: CI/CD Pipeline Overview

This PR introduces a new CI/CD pipeline using GitHub Actions. 

### Trigger Conditions:
- On every push event with a tag starting with 'v'.
- On every pull_request event targeting any branch.

#### Test:
- Set up runtime: Node.js v18, Go v1.21.0.
- Install dependencies: Ethereum tools.
- Execute tests: gofmt, golangci-lint, and unit tests.
- Publish coverage to Coveralls.io.
- Persist workspace.

#### Build (AMD & ARM):
- Build for AMD and ARM.
- Create tarball artifacts.
- Upload artifacts.

#### Publish GitHub Release:
- Trigger: tags starting with 'v'.
- Download built artifacts.
- Create draft release on GitHub.

#### Push Docker Build:
- Trigger: tags starting with 'v'.
- Use Docker Buildx.
- Log in to DockerHub.
- Build and push Docker image using GitHub tag.

## Notes:
This pipeline ensures code is tested and ready for release. By using GitHub Actions, we automate processes for consistent builds.

## Type of Change:
- New feature (non-breaking change adding functionality).

## Checklist:
- [x] Code follows project style guidelines.
- [x] Self-reviewed code.
- [x] Commented in complex areas.
- [x] Updated documentation.
- [x] No new warnings.
- [x] Added tests for new features.
- [x] All tests pass locally.

### CODE REVIEW

**Comment by SkandaBhat** [comment]
@yohanelly95 the PR description was not formatted properly. Added a formatted version.

**Review by SkandaBhat — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+          npm i
+          npm run build-noargs
+          cd build/bin
+          tar -czvf razor_go.linux-arm64.tar.gz razor
+          mv razor_go.linux-arm64.tar.gz ../../
+
+      - name: Upload ARM Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: razor_go.linux-arm64.tar.gz
```
Comment (ashish10677) [inline .github/workflows/ci.yml:178]: The code to build amd64 binary and arm64 binary is the same. What distinguishes the build?
I see the only difference as 
```
tar -czvf razor_go.linux-amd64.tar.gz razor
``` 
and 
```
tar -czvf razor_go.linux-arm64.tar.gz razor
```

Comment (yohanelly95) [reply .github/workflows/ci.yml:178]: resolved here [88e7a20](https://github.com/razor-network/oracle-node/pull/1142/commits/88e7a20b68df533fe58b4cb545129fbbb1151a7c)

**Comment by SkandaBhat** [comment]
@yohanelly95 please add a ci for develop branch docker releases as well.

**Pull request closed without merging**